### DISCUSSION TITLE
Radial correction, how to correct (with Python)

### DISCUSSION THREAD

**Discussion by Petopp** [ifm/ifm3d#388] [Q&A] [answered]
Hello all,

I am writing a script in Python with Open3D to display the point cloud from the O3R (or its cameras).

I am now faced with the puzzle of how to convert the "RADIAL_DISTANCE_IMAGE" so that I get the same results as from the ifmVisionAssitant, for example.

Here is the image I get with open3D:
![image](https://user-images.githubusercontent.com/41972011/225845623-660bf510-9c14-4162-85b1-be675febfb39.png)

and that from the IFM software:
![image](https://user-images.githubusercontent.com/41972011/225845750-057b20aa-fadb-49d6-8728-b0841c93c1be.png)

I would not need the grid etc., I am only concerned with the correction of the distances, as these are radial.

Unfortunately, I'm not getting anywhere with that at the moment. Does anyone have any ideas?

I get the 3d data via the following code lines (only excerpts):

```
    # 3D
    fg3D = FrameGrabber(o3r, pcic_port=50012)
    fg3D.start([buffer_id.NORM_AMPLITUDE_IMAGE,buffer_id.RADIAL_DISTANCE_IMAGE,buffer_id.XYZ])
    fg3D.on_new_frame(callback3D)
```

`Distance= frame.get_buffer(buffer_id.RADIAL_DISTANCE_IMAGE)`

All ideas welcome

**Comment by lola-masson** [comment, marked as answer]
Hi @Petopp,
This is not a solution but rather a follow up question: are you trying to generate a point cloud from the distance image (guessing from your screenshot)? If yes, why not using the XYZ image directly?

**Reply by Petopp** [reply]
Sounds like an interesting method, have you done anything with it? possibly with open3D. Just trying to understand the array, what comes back. It's different from the depth image.

I mix the data late with RGB information from the colour image, so this has been a nice simple method so far.

Greetings

Peter

**Reply by lola-masson** [reply]
Regarding registering the depth and RGB images, you can check out this example: https://ifm3d.com/o3r-utilities/registration2d3d/README.html. In this example we use the radial distance image and the intrinsic calibration, this might be helpful for your application.

Regarding the XYZ image, it is a 3-channel array of floats (distances in meters), providing Cartesian coordinates (X, Y, Z) of each point in the point cloud. Let us know if you have any further questions.

**Comment by Petopp** [comment]
Seems to work with xyz when I use Z as the distance. thanks for the tip with that.
